File: dataset_reduzido/codenet/Java/Code/s163827828.java
import java.util.*;
public class Main {
	public static void main(String args[]) {
		Scanner sc = new Scanner(System.in);
		int N = Integer.parseInt(sc.next());
		String s[] = new String[N];
		int a_num = 0;
		int b_num = 0;
		boolean flag = true;
		int ans = 0;
		for(int i = 0; i < N; i++) {
			s[i] = sc.next();
			if(s[i].substring(0, 1).equals("B")) {
				b_num++;
			}
			if(s[i].substring(s[i].length() - 1, s[i].length()).equals("A")){
				a_num++;
			}
			if(s[i].substring(0, 1).equals("B") || s[i].substring(s[i].length() - 1, s[i].length()).equals("A")) {
				if(!(s[i].substring(0, 1).equals("B") && s[i].substring(s[i].length() - 1, s[i].length()).equals("A"))) {
					flag = false;
				}
			}
			for(int j = 0; j < s[i].length() - 1; j++) {
				if(s[i].substring(j, j + 2).equals("AB")) ans++;
			}
		}
		ans += Math.min(a_num, b_num);
		if(Math.min(a_num, b_num) != 0 && flag) {
			ans--;
		}
		System.out.println(ans);
	}
}

File: output/Qwen/codenet/Java/Python/s163827828.py
N = int(input())
s = [input() for _ in range(N)]
a_num = sum(1 for x in s if x[-1] == "A")
b_num = sum(1 for x in s if x[0] == "B")
flag = all(x[0] == "B" and x[-1] == "A" for x in s)
ans = sum(1 for x in s for i in range(len(x)-1) if x[i:i+2] == "AB")
ans += min(a_num, b_num)
if(min(a_num, b_num) != 0 and flag):
    ans -= 1
print(ans)